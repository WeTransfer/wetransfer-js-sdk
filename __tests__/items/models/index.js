const { normalizeItem } = require('../../../src/items/models');

describe('Item model', () => {
  describe('normalizeItem function', () => {
    let item = {};
    beforeEach(() => {
      item = {
        filename: 'filename.txt',
        filesize: 1024,
        content_identifier: 'file',
        local_identifier: 'filename.txt'
      };
    });

    it('should return a normalized file item', () => {
      const file = normalizeItem(item);
      expect(file).toEqual(item);
    });

    it('should return a normalized link item', () => {
      item.content_identifier = 'web_content';
      const link = normalizeItem(item);
      expect(link).toEqual({
        content_identifier: 'web_content',
        local_identifier: 'filename.txt',
        meta: {
          title: ''
        },
        url: ''
      });
    });

    it('should remove extra properties', () => {
      const extraProps = Object.assign({}, item, { path: '/path/to/file.txt' });
      const file = normalizeItem(extraProps);
      expect(file).toEqual(item);
    });

    it('should throw an error if content_identifier is empty', () => {
      expect(() => normalizeItem({})).toThrowErrorMatchingSnapshot();
    });
  });
});
