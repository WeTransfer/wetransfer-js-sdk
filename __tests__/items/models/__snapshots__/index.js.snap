// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item model normalizeItem function should remove extra properties 1`] = `
Object {
  "content_identifier": "file",
  "filename": "filename.txt",
  "filesize": 1024,
  "local_identifier": "filename.txt",
}
`;

exports[`Item model normalizeItem function should return a normalized file item 1`] = `
Object {
  "content_identifier": "file",
  "filename": "filename.txt",
  "filesize": 1024,
  "local_identifier": "filename.txt",
}
`;

exports[`Item model normalizeItem function should return a normalized link item 1`] = `
Object {
  "content_identifier": "web_content",
  "local_identifier": "filename.txt",
  "meta": Object {
    "title": "",
  },
  "url": "",
}
`;

exports[`Item model normalizeItem function should throw an error if content_identifier is empty 1`] = `"Item's content_identifier property should be \\"file\\" or \\"web_content\\"."`;

exports[`Item model normalizeResponseItem function should normalize a file item 1`] = `
Object {
  "content_identifier": "file",
  "id": "random-hash",
  "local_identifier": "delightful-cat",
  "meta": Object {
    "multipart_parts": 3,
    "multipart_upload_id": "some.random-id--",
  },
  "name": "kittie.gif",
  "size": 195906,
}
`;

exports[`Item model normalizeResponseItem function should normalize a link item 1`] = `
Object {
  "content_identifier": "web_content",
  "id": "random-hash",
  "meta": Object {
    "title": "WeTransfer",
  },
}
`;
